tests/figures/chart/scorecard_chart_component.test.ts

```diff
@@ -273,57 +264,238 @@ describe("Scorecard charts", () => {
+    expect(chartDesign.baselineDescr?.style.color).toBeSameColorAs("#C8C8C8");
+    expect(chartDesign.key?.style.color).toBeSameColorAs("#FFFFFF");
+  });
+
+  test("Font size scale down if we put a long key value", () => {
+    createScorecardChart(
+      model,
+      { keyValue: "A1", baseline: "B2", title: "This is a title" },
+      chartId
+    );
+    const chartDesign1 = getChartDesign(model, chartId, sheetId);
+    setCellContent(model, "A1", "123456789123456789123456789");
+    const chartDesign2 = getChartDesign(model, chartId, sheetId);
+    expect(getContextFontSize(chartDesign2.baseline!.style.font)).toBeLessThan(
+      getContextFontSize(chartDesign1.baseline!.style.font)
+    );
+    expect(getContextFontSize(chartDesign2.title!.style.font)).toEqual(
+      getContextFontSize(chartDesign1.title!.style.font)
+    );
+    expect(getContextFontSize(chartDesign2.key!.style.font)).toBeLessThan(
+      getContextFontSize(chartDesign1.key!.style.font)
+    );
+  });
+
+  test("Font size scale up if we remove a long description", () => {
+    createScorecardChart(
+      model,
+      {
+        keyValue: "A1",
+        baseline: "B2",
+        baselineDescr: "This is a very very very very long description",
+      },
+      chartId
+    );
+    const chartDesign1 = getChartDesign(model, chartId, sheetId);
+    updateChart(model, chartId, { baselineDescr: "" }, sheetId);
+    const chartDesign2 = getChartDesign(model, chartId, sheetId);
+    expect(getContextFontSize(chartDesign2.baseline!.style.font)).toBeGreaterThan(
+      getContextFontSize(chartDesign1.baseline!.style.font)
+    );
+  });
+});
+
+describe("Scorecard charts rendering", () => {
+  beforeEach(() => {
+    chartId = "someuuid";
+    sheetId = "Sheet1";
+    const data = {
+      sheets: [
+        {
+          name: sheetId,
+          colNumber: 10,
+          rowNumber: 10,
+          rows: {},
+          cells: {
+            A1: { content: "2" },
+            A2: { content: "3" },
+            A3: { content: "2.1234" },
+            B1: { content: "1" },
+            B2: { content: "2" },
+            B3: { content: "3" },
+          },
+        },
+      ],
+    };
+    model = new Model(data);
+    scorecardChartStyle = {
+      title: {},
+      key: {},
+      baseline: {},
+      baselineDescr: {},
+    };
+    /*
+     * Be careful when writing new test using this, as the switch will mismatch elements
+     * that have the same text (ie baseline text = baselineDescr text = title text, ...).
+     * Be sure, when using this, to have all elements that are different in the test.
+     */
+    jest
+      .spyOn(MockCanvasRenderingContext2D.prototype, "fillText")
+      .mockImplementation(function (this: MockCanvasRenderingContext2D, text: string) {
+        const runtime = model.getters.getChartRuntime(chartId) as ScorecardChartRuntime;
+        switch (text) {
+          case runtime.baselineDisplay:
```

Comment:
I think you missed this one :) https://github.com/odoo/o-spreadsheet/pull/2968#discussion_r1361821451